atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/knowledge/KnownConcludedTransactionsStoreConcurrencyTest.java

```diff
@@ -86,6 +86,31 @@ public void batchesReadsUnderHighConcurrency() throws InterruptedException {
                 .get(eq(TransactionConstants.KNOWN_CONCLUDED_TRANSACTIONS_TABLE), anyMap());
     }
 
+    @Test
+    public void writesPreserveCorrectnessUnderHighConcurrencyAndMinimumTimestampSet() throws InterruptedException {
+        long minimumConcludableTimestamp = 10L;
+        int numThreads = 300;
+        List<Range<Long>> candidateTimestampRanges = LongStream.range(1, numThreads)
+                .mapToObj(index -> Range.closed(2 * index, 2 * index + 1))
+                .filter(range -> !range.isConnected(Range.closed(Long.MIN_VALUE, minimumConcludableTimestamp)))
+                .collect(Collectors.toList());
+
+        knownConcludedTransactionsStore.setMinimumConcludableTimestamp(10L);
+        startBlockingKeyValueServiceCalls();
+
+        List<Future<Void>> supplementFutures = scheduleTasksInParallel(numThreads, taskIndex -> {
+            knownConcludedTransactionsStore.supplement(candidateTimestampRanges.get(taskIndex));
+            return null;
+        });
+
+        letTasksRunToCompletion(supplementFutures, true);
+
+        Optional<ConcludedRangeState> rangesInDb = knownConcludedTransactionsStore.get();
+        assertThat(rangesInDb).hasValueSatisfying(timestampRangeSet -> assertThat(
+                        timestampRangeSet.timestampRanges().asRanges())
+                .isSubsetOf(candidateTimestampRanges));
```

Comment:
because of the name of the test, i was expecting an assert on the min timestamp.
but like i dont think we are even check the min bound update in this test?
maybe in this test we can solely have requests that extend the min bound and then see the request with highest minBounce that succeeded and assert on that.

Reply:
good idea that we assert that it has committed. I think we can rely on generating the expected subrange, which is effectively all ranges from [minTS, max], and per your earlier comment, we will submit ranges [1, max]. Thus this will show/prove that we do not supplement ranges below the minTs.